### PR TITLE
Sets up default country value in Billing Address

### PR DESCRIPTION
**Pull request by RafalOzogJudopay**

### CODE REVIEW

File: judokit-android/src/main/java/com/judopay/judokit/android/ui/cardentry/CardEntryViewModel.kt [hunk 1]
```diff
@@ -107,7 +107,7 @@ class CardEntryViewModel(
 
     // used when the form needs to be pre populated, ex. `Scan Card`
     private var cardDetailsModel = CardDetailsInputModel()
-    private var billingAddressModel = BillingDetailsInputModel()
+    private var billingAddressModel = BillingDetailsInputModel(context)
```
Comment (RafalOzogJudopay) [inline judokit-android/src/main/java/com/judopay/judokit/android/ui/cardentry/CardEntryViewModel.kt:110]: Passing context is not great (I agree), but explained below.

File: judokit-android/src/main/java/com/judopay/judokit/android/ui/cardentry/components/BillingDetailsFormView.kt [hunk 2]
```diff
@@ -197,8 +197,16 @@ class BillingDetailsFormView
...
-                        setText(this)
-                        textDidChange(type, this, FormFieldEvent.TEXT_CHANGED)
+                        var value = this
+                        if (type == BillingDetailsFieldType.COUNTRY) {
+                            if (value.isNotEmpty()) {
+                                countries.find { it.numericCode.equals(value, ignoreCase = true) }?.name?.let {
+                                    value = it
+                                }
+                            }
+                        }
```
Comment (RafalOzogJudopay) [inline judokit-android/src/main/java/com/judopay/judokit/android/ui/cardentry/components/BillingDetailsFormView.kt:207]: This is purely to setup UI correctly. In all other fields we can just set text value, and it's gonna match the input. But in country case we get countryNumericCode from the model ('US'), so we want to turn it into 'United States' (that would be there if someone chooses it from list). Tested.

File: judokit-android/src/main/java/com/judopay/judokit/android/ui/cardentry/model/BillingDetailsInputModel.kt [hunk 3]
```diff
@@ -17,4 +19,8 @@ data class BillingDetailsInputModel(
     var submitButtonState: ButtonState = ButtonState.Disabled(R.string.pay_now),
     var backButtonState: ButtonState = ButtonState.Enabled(R.string.back),
     var isValid: Boolean = false,
-)
+) {
+    constructor(context: Context) : this(
+        countryCode = getDefaultCountry(context)?.numericCode.orEmpty()
+    )
```
Comment (RafalOzogJudopay) [inline judokit-android/src/main/java/com/judopay/judokit/android/ui/cardentry/model/BillingDetailsInputModel.kt:25]: I tried various solutions of setting this default value. This one, despite passing context to the model, seems optimal to me.

We need context to set up the default value (either way using TelephonyManager or Locals, as both require countries list that also needs context). Unless we want to pass the whole countries list (then we could make context optional, but it doesn't change much for this code).

So as we need context to find default country, we can set it up from the outside (I tried to set it up from CardEntryViewModel, but then it's not updated early enough, can explain more if needed), or from the inside (but then we need to pass context).

What do you think @stefan-tudor , any suggestions?

**Comment by stefan-tudor** [comment]
covered by https://github.com/Judopay/JudoKit-Android/pull/265

**Pull request closed without merging**